rpmio/rpmlog.cc

```diff
@@ -412,3 +415,43 @@ void rpmlog (int code, const char *fmt, ...)
 exit:
     errno = saved_errno;
 }
+
+int rpmlogOnce (uint64_t domain, const char * key, int code, const char *fmt, ...)
+{
+    int saved_errno = errno;
+    rpmlogCtx ctx = rpmlogCtxAcquire();
+    int newkey = 0;
+
+    if (ctx) {
+	wrlock lock(ctx->mutex);
+	newkey = !ctx->seen[domain][{code, key}]++;
+    }
+
+    if (newkey) {
+	va_list ap;
+	char *msg = NULL;
+	va_start(ap, fmt);
+	if (rvasprintf(&msg, fmt, ap) >= 0) {
+	    rpmlog(code, msg);
+	    free(msg);
+	}
+	va_end(ap);
+    }
+    errno = saved_errno;
+    return newkey;
+}
+
+void rpmlogReset(uint64_t domain, int mode=0)
+{
+    rpmlogCtx ctx = rpmlogCtxAcquire();
+    std::map<std::pair<int, std::string>, int> domain_data = {};
+
+    if (ctx) {
+	wrlock lock(ctx->mutex);
+	if (mode)
+	    domain_data = ctx->seen[domain];
```

Comment:
Don't add speculative dead code, this only looks baffling. Add handling for other modes when you add them.

Reply:
Deleted.